src/drivers/net/rtl8139.rs

```diff
@@ -277,10 +279,28 @@ impl NetworkInterface for RTL8139Driver {
 
 			if header & ISR_ROK == ISR_ROK {
 				let length = self.rx_peek_u16() - 4; // copy packet (but not the CRC)
-				let buf = &self.rxbuffer[self.rxpos + mem::size_of::<u16>()..][..length.into()];
+				let pos = (self.rxpos + mem::size_of::<u16>()) % RX_BUF_LEN;
+
+				// do we reach the end of the receive buffers?
+				// in this case, we have to copy the data in boxed slice
+				let buf = if pos + length as usize > RX_BUF_LEN {
+					let mut msg: Box<[u8]> = vec![0; length as usize].into_boxed_slice();
+					let limit = RX_BUF_LEN - pos;
+
+					msg[0..limit].copy_from_slice(&self.rxbuffer[pos..RX_BUF_LEN]);
+					msg[limit..].copy_from_slice(&self.rxbuffer[0..length as usize - limit]);
```

Comment:
We can also avoid to zero the buffer by using `[T]::concat`:

```suggestion
					let first = &self.rxbuffer[pos..RX_BUF_LEN];
					let second = &self.rxbuffer[..length as usize - first.len()];
					let msg = [first, second].concat().into_boxed_slice();
```